dbms/src/Operators/SharedAggregateRestorer.cpp

```diff
@@ -48,6 +48,22 @@ bool SharedSpilledBucketDataLoader::switchStatus(SharedLoaderStatus from, Shared
     return status.compare_exchange_strong(from, to);
 }
 
+bool SharedSpilledBucketDataLoader::checkCancelled()
+{
+    if unlikely (exec_context.isCancelled())
+    {
+        {
+            std::lock_guard lock(mu);
+            if (is_cancelled)
```

Comment:
looks like use cas here is enough?

Reply:
https://github.com/pingcap/tiflash/pull/9082/files#diff-b54d5b998f3a9de97d9452c7c40a8016e79ecbc2251784e96ac0fd6ee840fa16R159-R163
It's to ensure SharedSpilledBucketDataLoader::registerTask can immediately notify the task when cancelled, rather than moving the task to pipe_read_cv.

Reply:
Otherwise, it may cause the task to be permanently in pipe_read_cv, unable to be notified when the query is cancelled.

Reply:
I mean if you use a atomicbool for `is_cancelled`, then you don't need to acquire lock here